ignite/engine/engine.py

```diff
@@ -621,6 +623,9 @@ def run(
                 `len(data)`. If `data` is an iterator and `epoch_length` is not set, then it will be automatically
                 determined as the iteration on which data iterator raises `StopIteration`.
                 This argument should not change if run is resuming from a state.
+            max_iters (int, optional): Max Number of iterations to run for. By default, it will be calculated as
+                max_epochs * epoch_length. If `data` is an iterator and `max_iters` is not set, then it will be
+                automatically determined along with `epoch_length` on which data iterator raises `StopIteration`.
```

Comment:
Let's also mention that `max_iters` and `max_epochs` are mutually exclusive args and add this check inside `run` function.